api/src/neo4j/queries/search/globalSearch.js

```diff
@@ -18,14 +18,22 @@ const fetchCompartmentalizedMetabolites = async ({
   if (viaMetabolties) {
     statement += `
 WITH ${JSON.stringify(ids)} as mids
-UNWIND mids as mid
+UNWIND
+  CASE
+      WHEN mids = [] THEN [null]
+      ELSE mids
+  END AS mid
 MATCH (:Metabolite:${model} {id:mid})-[${version}]-(cm:CompartmentalizedMetabolite)
 WITH DISTINCT(cm.id) as cmid
 `;
   } else {
     statement += `
 WITH ${JSON.stringify(ids)} as cmids
-UNWIND cmids as cmid
+UNWIND
+  CASE
+      WHEN cmids = [] THEN [null]
+      ELSE cmids
+  END AS cmid
```

Comment:
Did this cause any problems specifically for the global search? Just curious since it seemed like it wasn't needed for the model search.

Reply:
There was no error what I noticed in global search, but it felt safer to add this if (c)mids would be empty for some query.

Reply:
I don't think we tested anything in global search? Either way, I also agree it makes sense to keep it as is for now

Reply:
You are right! I assumed that it would work the same way.